docs/eessi/high-level-design.md

```diff
@@ -0,0 +1,241 @@
+
+```
+$ ls software/linux/x86_64/amd/zen3
+modules  software
+
+$ ls software/linux/x86_64/amd/zen3/software
+
+... (long list of directories of software names omitted) ...
+
+$ ls software/linux/x86_64/amd/zen3/software/OpenBLAS/
+0.3.21-GCC-12.2.0  0.3.23-GCC-12.3.0
+
+$ ls software/linux/x86_64/amd/zen3/software/OpenBLAS/0.3.23-GCC-12.3.0/
+bin  easybuild	include  lib  lib64
+
+$ ls software/linux/x86_64/amd/zen3/modules/all
+
+... (long list of directories of software names omitted) ...
+
+$ ls software/linux/x86_64/amd/zen3/modules/all/OpenBLAS
+0.3.21-GCC-12.2.0.lua  0.3.23-GCC-12.3.0.lua
+```
+
+Each of the other subdirectories for specific CPU microarchitectures will have the exact same structure,
+and provide the same software installations and accompanying environment module files to access them with Lmod.
+
+A key aspect here is that binaries and libraries that make part of the software installations
+included in the EESSI software layer only rely on libraries provided by the compatibility layer.
+
+See for example libraries to which the OpenBLAS library links:
+
+```
+$ ldd software/linux/x86_64/amd/zen3/software/OpenBLAS/0.3.23-GCC-12.3.0/lib/libopenblas.so
+	linux-vdso.so.1 (0x00007ffd4373d000)
+	libm.so.6 => /cvmfs/software.eessi.io/versions/2023.06/compat/linux/x86_64/lib/../lib64/libm.so.6 (0x000014d0884c8000)
+	libgfortran.so.5 => /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/amd/zen3/software/GCCcore/12.3.0/lib64/libgfortran.so.5 (0x000014d087115000)
+	libgomp.so.1 => /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/amd/zen3/software/GCCcore/12.3.0/lib64/libgomp.so.1 (0x000014d088480000)
+	libc.so.6 => /cvmfs/software.eessi.io/versions/2023.06/compat/linux/x86_64/lib/../lib64/libc.so.6 (0x000014d086f43000)
+	/lib64/ld-linux-x86-64.so.2 (0x000014d08837e000)
+	libpthread.so.0 => /cvmfs/software.eessi.io/versions/2023.06/compat/linux/x86_64/lib/../lib64/libpthread.so.0 (0x000014d088479000)
+	libdl.so.2 => /cvmfs/software.eessi.io/versions/2023.06/compat/linux/x86_64/lib/../lib64/libdl.so.2 (0x000014d088474000)
+	libquadmath.so.0 => /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/amd/zen3/software/GCCcore/12.3.0/lib64/libquadmath.so.0 (0x000014d08842d000)
+	libgcc_s.so.1 => /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/amd/zen3/software/GCCcore/12.3.0/lib64/libgcc_s.so.1 (0x000014d08840d000)
+```
+
+??? note "Note on `/lib64/ld-linux-x86-64.so.2` *(click to expand)*"
+
+    The `/lib64/ld-linux-x86-64.so.2` path, which corresponds to the dynamic linker/loader of the Linux client OS,
+    that is shown in the output of `ldd` above is a bit misleading.
+
+    It only pops up because we are running the `ldd` command provided by the client OS, which typically
+    resides at `/usr/bin/ldd`.
+
+    When actually running software provided by the EESSI software layer, the loader provided
+    by the EESSI compatibility layer is used to launch binaries.
```

Comment:
Sure, but that's not really the actual user experience.

End users should just run the init script (which also updates `$PATH` to pick up tools from compat layer), not run `startprefix`.

I consider this a detail (which is why I made it a collapsed note), but I often find myself using the wrong `ldd` and then scratching my head because there's a client OS path in the output, so I figured I would try to clarify it...